cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -849,8 +849,8 @@ public ValidationResult validateNetworkLinkExistenceForDnsZones(String networkLi
         for (AzurePrivateDnsZoneServiceEnum service : services) {
             String dnsZoneName = service.getDnsZoneName();
             Optional<PrivateZone> privateZoneWithNetworkLink = privateDnsZoneList.stream()
-                    .filter(privateZone -> !privateZone.resourceGroupName().equals(resourceGroupName))
-                    .filter(privateZone -> privateZone.name().equals(dnsZoneName))
+                    .filter(privateZone -> !privateZone.resourceGroupName().equalsIgnoreCase(resourceGroupName))
```

Comment:
AzureClientTest ?

Reply:
This contains merely the fix, there is a refactor in progress (on branch network-link-validation-fix) to be done that contains tests covering the various cases